lib/logstash/filters/dns.rb

```diff
@@ -169,19 +167,9 @@ def resolve(event)
       end
 
       if @action == "replace"
-        if is_array
-          event.set(field, [address])
-        else
-          event.set(field, address)
-        end
+        event.set(field, address)
```

Comment:
When an event has a single-element array, before this change we would replace that array with a single-element array; after this change, we would replace it with a single string. Although many outputs like Elasticsearch treat these as functionally-equivalent, it is not always the case so I am hesitant to accept this change as-is.

Reply:
I see what you mean now, thanks for the explanation.  Feel free to close the PR.